src/concrete/ml/quantization/quantized_module.py

```diff
@@ -561,30 +631,52 @@ def _fhe_forward(
             return q_results[0]
         return q_results
 
-    def quantize_input(self, *x: numpy.ndarray) -> Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]:
+    def quantize_input(
+        self, *x: Optional[numpy.ndarray]
+    ) -> Union[numpy.ndarray, Tuple[Optional[numpy.ndarray], ...]]:
         """Take the inputs in fp32 and quantize it using the learned quantization parameters.
 
         Args:
-            x (numpy.ndarray): Floating point x.
+            x (Optional[numpy.ndarray]): Floating point x or None.
 
         Returns:
-            Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]: Quantized (numpy.int64) x.
+            Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]: Quantized (numpy.int64) x, or None if
+                the corresponding input is None.
         """
         n_inputs = len(self.input_quantizers)
         n_values = len(x)
+
         assert_true(
             n_values == n_inputs,
             f"Got {n_values} inputs, expected {n_inputs}. Either the quantized module has not been "
             "properly initialized or the input data has been changed since its initialization.",
             ValueError,
         )
 
-        q_x = tuple(self.input_quantizers[idx].quant(x[idx]) for idx in range(len(x)))
+        assert not all(x_i is None for x_i in x), "Please provide at least one input to quantize."
+
+        # Ignore [arg-type] check from mypy as it is not able to see that the input to `quant`
+        # cannot be None
+        q_x = tuple(
```

Comment:
these are needed to match how CP works with encrypt, ie `encrypt(None, x) = None, x_enc`, since we do not encrypt all inputs at the same time with composition